src/api/c/graphics_common.cpp

```diff
@@ -0,0 +1,188 @@
+/*******************************************************
+ * Copyright (c) 2014, ArrayFire
+ * All rights reserved.
+ *
+ * This file is distributed under 3-clause BSD license.
+ * The complete license agreement can be obtained at:
+ * http://arrayfire.com/licenses/BSD-3-Clause
+ ********************************************************/
+
+#if defined(WITH_GRAPHICS)
+
+#include <graphics_common.hpp>
+#include <err_common.hpp>
+#include <backend.hpp>
+#include <platform.hpp>
+
+using namespace std;
+
+template<typename T>
+GLenum getGLType() { return GL_FLOAT; }
+
+#define INSTANTIATE_GET_GL_TYPE(T, OpenGLEnum)\
+    template<> GLenum getGLType<T>() { return OpenGLEnum; }
+
+INSTANTIATE_GET_GL_TYPE(float, GL_FLOAT);
+INSTANTIATE_GET_GL_TYPE(int  , GL_INT);
+INSTANTIATE_GET_GL_TYPE(unsigned, GL_UNSIGNED_INT);
+INSTANTIATE_GET_GL_TYPE(char, GL_BYTE);
+INSTANTIATE_GET_GL_TYPE(unsigned char, GL_UNSIGNED_BYTE);
+
+GLenum glErrorSkip(const char *msg, const char* file, int line)
+{
+#ifndef NDEBUG
+    GLenum x = glGetError();
+    if (x != GL_NO_ERROR) {
+        printf("GL Error Skipped at: %s:%d Message: %s Error Code: %d \"%s\"\n", file, line, msg, x, gluErrorString(x));
+    }
+    return x;
+#else
+    return 0;
+#endif
+}
+
+GLenum glErrorCheck(const char *msg, const char* file, int line)
+{
+// Skipped in release mode
+#ifndef NDEBUG
+    GLenum x = glGetError();
+
+    if (x != GL_NO_ERROR) {
+        printf("GL Error at: %s:%d Message: %s Error Code: %d \"%s\"\n", file, line, msg, x, gluErrorString(x));
```

Comment:
I didn't quite understand what you refer to here. Should i add AF_ERROR_GL_ERR to af_err enum ?